### ISSUE TITLE
Poor performance when using Ingress gateway with HTTPS

### ISSUE THREAD

**Issue by liquid-cloud** [istio/istio#21025]
**Bug description**

Pasted the test results here: https://github.com/liquid-cloud/hello-go/blob/781af1a7cf0b183923b32b091ec51377e2aaa987/istio_results.md

**Expected behavior**

The expected behavior would be similar or closer performance to the nginx ingress, but from what it looks it's ~4 times slower.

**Steps to reproduce the bug**

The manifests for the application are here: https://github.com/liquid-cloud/hello-go/tree/master/kube-manifests

Note that it’s one single application exposed with both Istio and Nginx using HTTP/HTTPS.

You should also be able to reproduce the issue by deploying the yamls in any cluster.

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

**How was Istio installed?**

Installed with Rancher, it uses the helm chart from here: https://github.com/rancher/system-charts/tree/dev/charts/rancher-istio/1.4.300

**Environment where bug was observed (cloud vendor, OS, etc)**

**Comment by howardjohn** [comment]
I reproduced this using your configs, but with `httpbin` app since that one is not public. Some notes.

* Your usage of siege seems invalid - you are trying to measure throughput but injecting artificial delay? see `  -b, --benchmark           BENCHMARK: no delays between requests.` flag
* You have httpsRedirect, so HTTPS being slower is highly suspicious
* Using `hey` I see https is 2x faster
* if you run siege with `-v` it shows some clearly invalid data:
```
HTTP/1.1 301     0.11 secs:       0 bytes ==> GET  /get
HTTP/1.1 200     0.21 secs:     803 bytes ==> GET  /get
HTTP/1.1 301     0.11 secs:       0 bytes ==> GET  /get
HTTP/1.1 200     0.30 secs:     805 bytes ==> GET  /get
HTTP/1.1 301     0.11 secs:       0 bytes ==> GET  /get
HTTP/1.1 200     0.25 secs:     803 bytes ==> GET  /get
HTTP/1.1 301     0.11 secs:       0 bytes ==> GET  /get
HTTP/1.1 200     0.26 secs:     805 bytes ==> GET  /get
HTTP/1.1 301     0.10 secs:       0 bytes ==> GET  /get
HTTP/1.1 200     0.25 secs:     803 bytes ==> GET  /get

Lifting the server siege...
Transactions:                     10 hits
Availability:                 100.00 %
Elapsed time:                   9.90 secs
Data transferred:               0.00 MB
Response time:                  0.18 secs
```

The fastest response time is 0.32s but it says the average is 0.18...

I also tried removing the httpsRedirect, in which case I see https ~5% slower, which is expected due to tls overhead

More info: https://istio.io/latest/blog/2019/performance-best-practices/

tl;dr I strongly suspect siege is reporting inaccurate measurements.

**Comment by howardjohn** [comment]
Closing as I don't think this is a bug in Istio and there hasn't been any followup. Thanks!!
